src/lib/forms/widgets/loader.js

```diff
@@ -8,30 +8,37 @@ import React from "react";
  * dynamic import cannot rely on purely a dynamic path i.e a variable.
  */
 export async function importWidget(
-  templateLoader,
+  templateLoaders,
   { ui_widget: UIWidget, fieldPath, props }
 ) {
-  let component = null;
-  try {
-    // First try import widget from user's defined templateLoader
-    const module = await templateLoader(UIWidget);
-    component = module.default ?? module[UIWidget];
-  } catch (error) {
+  let component = undefined;
+
+  // Try import widget from user's defined templateLoaders
+  for await (const loader of templateLoaders) {
```

Comment:
to be checked but you could do
```suggestion
  templateLoaders.some(async loader => {
      const module = await loader(UIWidget);
      component = module.default ?? module[UIWidget];
      return component !== undefined
  })
```
`some` exits in the first truthy evaluation so you could avoid the if block.... See [some docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some#description)

Reply:
As discussed IRL, `some` does not guarantee order and we need it here.

Therefore, I will simply use iterate the loaders with a `for ... of` and await the result. When the first one works, the loop breaks.